src/Toolkit/Html.php

```diff
@@ -446,128 +447,110 @@ public static function video(string $url, array $options = [], array $attr = [])
      */
     public static function vimeo(string $url, array $options = [], array $attr = []): string
     {
-        if (preg_match('!vimeo.com\/([0-9]+)!i', $url, $array) === 1) {
-            $id = $array[1];
-        } elseif (preg_match('!player.vimeo.com\/video\/([0-9]+)!i', $url, $array) === 1) {
-            $id = $array[1];
-        } else {
+        $uri   = new Uri($url);
+        $path  = $uri->path();
+        $query = $uri->query();
+        $id    = null;
+
+        switch ($uri->host()) {
+            case 'vimeo.com':
+            case 'www.vimeo.com':
+                $id = $path->first();
+                break;
+            case 'player.vimeo.com':
+                $id = $path->nth(1);
+                break;
+        }
+
+        if (!preg_match('!^[0-9]*$!', $id)) {
             throw new Exception('Invalid Vimeo source');
         }
 
-        // build the options query
-        if (empty($options) === false) {
-            $query = '?' . http_build_query($options);
-        } else {
-            $query = '';
+        // append query params
+        foreach ($options as $key => $value) {
+            $query->$key = $value;
         }
 
-        $url = 'https://player.vimeo.com/video/' . $id . $query;
+        $src = 'https://player.vimeo.com/video/' . $id . $query->toString(true);
 
-        return static::iframe($url, array_merge(['allowfullscreen' => true], $attr));
+        return static::iframe($src, array_merge(['allowfullscreen' => true], $attr));
     }
 
     /**
      * Embeds a YouTube video by URL in an `<iframe>`
      *
      * @param string $url YouTube video URL
      * @param array $options Query params for the embed URL
-     * @param array $attr Additional attributes for the `<iframe>` tag
+     * @param array $attrs Additional attributes for the `<iframe>` tag
      * @return string The generated HTML
      */
-    public static function youtube(string $url, array $options = [], array $attr = []): string
+    public static function youtube(string $url, array $options = [], array $attrs = []): string
```

Comment:
Renaming the param would be a breaking-change because of PHP 8's named arguments. Also it's only there for the YouTube method, not for Vimeo. Is the change really necessary?

Reply:
Ah, I still have to get used to the PHP 8 stuff. No, it's not necessary. I will revert it.